projects/packages/stats/src/class-options.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Stats Options
+ *
+ * @package automattic/jetpack-stats
+ */
+
+namespace Automattic\Jetpack\Stats;
+
+use Automattic\Jetpack\Constants;
+use Jetpack_Options;
+
+/**
+ * Stats Options class.
+ */
+class Options {
+
+	/**
+	 * Option name.
+	 *
+	 * @var string $option_name The 'stats' option name
+	 */
+	public static $option_name = 'stats_options';
+
+	/**
+	 * Stats Options.
+	 *
+	 * @var array $options An array of associated Stats options (default empty)
+	 */
+	protected static $options = array();
+
+	/**
+	 * Stats Get Options.
+	 *
+	 * @return array.
+	 */
+	public static function get_options() {
+		// Make sure we only get options from the database once per connection.
+		if ( count( self::$options ) > 0 ) {
+			return self::$options;
+		}
+
+		self::$options = get_option( self::$option_name, array() );
+		self::$options = array_merge( self::get_defaults(), self::$options );
+
+		if ( self::$options['version'] < Constants::get_constant( 'STATS_VERSION' ) ) {
+			self::upgrade_options( self::$options );
+		}
+
+		return self::$options;
+	}
+
+	/**
+	 * Get Stats Option..
+	 *
+	 * @param string $option Option name.
+	 * @return mixed|null.
+	 */
+	public static function get_option( $option ) {
+		if ( 'blog_id' === $option ) {
+			return Jetpack_Options::get_option( 'id' );
+		}
+
+		$options = self::get_options();
+
+		if ( isset( $options[ $option ] ) ) {
+			return $options[ $option ];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Stats Set Option.
+	 *
+	 * @param string $option The option name.
+	 * @param mixed  $value The option Value.
+	 * @return bool.
+	 */
+	public static function set_option( $option, $value ) {
+		$options = self::get_options();
+
+		$options[ $option ] = $value;
+
+		return self::set_options( $options );
+	}
+
+	/**
+	 * Stats Set Options.
+	 *
+	 * @access public
+	 * @param array $options Options.
+	 * @return bool
+	 */
+	public static function set_options( $options ) {
+		if ( ! is_array( $options ) ) {
+			return false;
+		}
+
+		$defaults       = self::get_defaults();
+		$stored_options = get_option( self::$option_name, array() );
+		$all_options    = array_merge( $defaults, $stored_options );
+		$options        = array_merge( $all_options, $options );
+
+		$allowed_options = array_keys( $defaults );
+		foreach ( $options as $option_name => $option_value ) {
+			if ( ! in_array( $option_name, $allowed_options, true ) ) {
+				unset( $options[ $option_name ] );
+			}
+		}
+
+		$options['blog_id'] = Jetpack_Options::get_option( 'id' );
+		$options['version'] = Constants::get_constant( 'STATS_VERSION' );
+
+		$success = update_option( 'stats_options', $options );
```

Comment:
Shouldn't we be using `self::$option_name` here as well?

Reply:
🤦